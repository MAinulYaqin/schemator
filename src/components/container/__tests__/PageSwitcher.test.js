import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";

import ConnectedComponent, { PageSwitcher } from "../PageSwitcher";

jest.mock("../WorkArea", () => "WorkArea");
jest.mock("../Extensions", () => "Extensions");

describe("PageSwitcher", () => {
  const setup = propOverrides => {
    const props = {
      page: null,
      ...propOverrides
    };

    const wrapper = shallow(<PageSwitcher {...props} />);

    return {
      wrapper,
      props
    };
  };

  const setupConnected = () => {
    const mockStore = configureStore();
    const store = mockStore({
      page: "workarea"
    });

    store.dispatch = jest.fn();

    const wrapper = shallow(<ConnectedComponent store={store} />);

    return {
      wrapper,
      store
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders as specified page", () => {
    const pages = ["workarea", "extensions"];

    pages.forEach(page => {
      const { wrapper } = setup({ page });
      expect(wrapper).toMatchSnapshot();
    });
  });

  it("should maps state to props", () => {
    const { wrapper, store } = setupConnected();
    const { page } = store.getState();

    expect(wrapper.props()).toMatchObject({ page });
  });
});
