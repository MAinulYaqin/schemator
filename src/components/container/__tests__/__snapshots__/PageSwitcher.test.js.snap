// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSwitcher should renders as specified page 1`] = `<WorkArea />`;

exports[`PageSwitcher should renders as specified page 2`] = `<Extensions />`;

exports[`PageSwitcher should renders default 1`] = `<WorkArea />`;
